### DISCUSSION TITLE
Dependency on vendored gems in gemspec

### DISCUSSION THREAD

**Discussion by svoop** [ruby/rubygems#6634] [Q&A] [answered]
I'm trying to build a gem which contains vendored gems. The idea: Non-essential parts of the gem functionality are extracted into plugins (think: Roda) and each plugin is wrapped as a vendored gem. Why? It's easier to later extract plugins into separate gems if need be. (And it solves some naming issues as well.)

When doing this in a Ruby project, it works as expected. However, when I move the vendored gems inside the Ruby gem project, I hit a wall.

[Here's an example gem](https://github.com/svoop/mygem) which is just a vanilla gem (created with `bundle gem`) and two alterations:

* a vendored vanilla gem (created with `bundle gem`) inside `vendor/gems/myvendoredgem`
* the main gem [depends on the vendored gem](https://github.com/svoop/mygem/blob/f5204f1b8dc50b8989f057d4da5fedbf634b7dfa/mygem.gemspec#L28)

`bundle install` doesn't like this:

> Because every version of mygem depends on myvendoredgem >= 0
and myvendoredgem >= 0 could not be found in rubygems repository
https://rubygems.org/ or installed locally,

That's not correct, at least not if "installed locally" includes those inside `vendor/gems`.

Is it possible to get this to work?

**Comment by pboling** [comment, marked as answer]
You have to specify the path in the Gemfile.
